### DISCUSSION TITLE
Elsa 3.1 - Get previous activity result

### DISCUSSION THREAD

**Discussion by rmanjusha** [elsa-workflows/elsa-core#5251] [Q&A]
How to access previous activity result in If branch condition ?

**Comment by glime-ai** [comment]
To access the result of a previous activity in an If branch condition within Elsa workflows, you can use the `Outcome` property of the previous activity or access the workflow context to retrieve the output of the previous activity. Here's an example of how you might do this:

```csharp
public class MyWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            // ... other activities ...
            .Then<PreviousActivity>() // This is the activity whose output you want to use
            .If(context => context.GetLastResult<bool>()) // Accessing the result of the previous activity
                .Then<ActivityIfTrue>() // Activities to execute if the condition is true
            .Else()
                .Then<ActivityIfFalse>(); // Activities to execute if the condition is false
    }
}
```

In the example above, `context.GetLastResult<bool>()` is used to retrieve the output of the `PreviousActivity`. This assumes that the output of `PreviousActivity` is a boolean value which is then used as the condition for the If activity.

Please note that the exact method to access the previous activity's result may vary depending on the version of Elsa you are using and the specifics of your workflow. If you are using Elsa 2.0 or later, the approach shown should be applicable.

If you need further clarification or if your scenario is different, please provide more details about your workflow and the activities you are using.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by rmanjusha** [reply]
How to access the result of a previous activity in an If branch condition in studio designer (not from code) ?
